### PR TITLE
Upgrade to helm v3 libraries

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

File: licenses/github.com/xeipuuv/gojsonpointer/NONE [hunk 1]
```diff
@@ -0,0 +1 @@
+NO LICENSE FOUND
```
Comment (ostromart) [inline licenses/github.com/xeipuuv/gojsonpointer/NONE:1]: is this a problem for us?

**Comment by howardjohn** [comment]
Blocked by https://github.com/istio/istio/pull/23131, will automagically be fixed when that is merged

File: operator/pkg/helm/fs_renderer.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ type FileTemplateRenderer struct {
 // NewFileTemplateRenderer creates a TemplateRenderer with the given parameters and returns a pointer to it.
 // helmChartDirPath must be an absolute file path to the root of the helm charts.
 func NewFileTemplateRenderer(helmChartDirPath, componentName, namespace string) *FileTemplateRenderer {
-	scope.Infof("NewFileTemplateRenderer with helmChart=%s, componentName=%s", helmChartDirPath, componentName)
+	scope.Debugf("NewFileTemplateRenderer with helmChart=%s, componentName=%s", helmChartDirPath, componentName)
```
Comment (ostromart) [inline operator/pkg/helm/fs_renderer.go:36]: i'd leave this as info or remove if it's not really useful. debug scope is going to be reserved for voluminous dumps like objects.

File: operator/pkg/helm/helm.go [hunk 3]
```diff
@@ -25,14 +25,17 @@ import (
...
-	"k8s.io/helm/pkg/proto/hapi/chart"
-	"k8s.io/helm/pkg/timeconv"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/engine"
+	"sigs.k8s.io/yaml"
+
+	"helm.sh/helm/v3/pkg/chartutil"
+
+	"istio.io/pkg/log"
```
Comment (ostromart) [inline operator/pkg/helm/helm.go:35]: format

File: licenses/github.com/xeipuuv/gojsonpointer/NONE (hunk 1 above)
Comment (howardjohn) [reply licenses/github.com/xeipuuv/gojsonpointer/NONE:1]: It is. We have a manual whitelist process. This does have a license its just not found by our linter due to a weird name - which is why this PR fails until common files updates, where i updated the whitelist

File: operator/pkg/helm/helm.go (hunk 3 above)
Comment (howardjohn) [reply operator/pkg/helm/helm.go:35]: I am not sure what you are looking for here? This is run through `make format`

Comment (ostromart) [reply operator/pkg/helm/helm.go:35]: make format doesn't merge blocks separated by blank lines - just remove the two blank lines and reformat

Comment (howardjohn) [reply operator/pkg/helm/helm.go:35]: I'll update but I think if we want to enforce style issues like this they should be encoded into `make format`/`make lint`

Comment (nrjpoddar) [reply operator/pkg/helm/helm.go:35]: Enforcement >>>> Code review convention

**Review by nrjpoddar — APPROVED** [review]
@linsun this needs to go in for other work related to legacy migrations to start. Thanks!

**Review by ostromart — APPROVED** [review]

**Review by linsun — APPROVED** [review]

**Pull request merged**